geoh5py/objects/curve.py

```diff
@@ -115,30 +83,29 @@ def current_line_id(self, value: uuid.UUID | None):
             )
 
         self._current_line_id = value
-        self.workspace.update_attribute(self, "attributes")
 
-    @classmethod
-    def default_type_uid(cls) -> uuid.UUID:
+        if self.on_file:
+            self.workspace.update_attribute(self, "attributes")
+
+    def _make_cells_from_parts(self) -> np.ndarray:
         """
-        :return: Default unique identifier
+        Generate cells from parts.
         """
-        return cls.__TYPE_UID
+        cells = []
```

Comment:
```suggestion

cells = np.c_[ np.arange(self.n_vertices-1),  np.arange(1, self.n_vertices)]
indices_to_suppress = np.flatnonzero(np.diff(self.parts)) + 1
cells = np.delete(cells, indices_to_suppress, axis=0)
return cells

Reply:
removing loops ;-)

Reply:
vertices might not be ordered, so we need to fetch each part sequentially.